File: exemple.java
//import required libraries
import discord4j.core.DiscordClient;
import discord4j.core.event.domain.message.MessageCreateEvent;
import discord4j.core.object.entity.Message;
import discord4j.core.object.entity.channel.TextChannel;

//create the main class
public class MyDiscordBot {

   //declare the Discord client and the prefix to use for commands
   private static final String PREFIX = "!";
   private final DiscordClient client;

   //constructor to initialize the client
   public MyDiscordBot(String token) {
      this.client = DiscordClient.create(token);
   }

   //method to start the bot
   public void start() {
      client.login().block();
      client.getEventDispatcher().on(MessageCreateEvent.class)
            .map(MessageCreateEvent::getMessage)
            .filter(message -> message.getAuthor().map(user -> !user.isBot()).orElse(false))
            .filter(message -> message.getContent().startsWith(PREFIX))
            .flatMap(Message::getChannel)
            .ofType(TextChannel.class)
            .flatMap(channel -> channel.createMessage("Hello, I'm a Discord bot!"))
            .subscribe();
   }

   //main method to run the bot
   public static void main(String[] args) {
      MyDiscordBot bot = new MyDiscordBot("your-bot-token-here");
      bot.start();
   }
}

File: exemple.kt
import discord4j.core.DiscordClientBuilder
import discord4j.core.`object`.entity.Message

fun main() {
    val token = "your-bot-token"
    val client = DiscordClientBuilder(token).build()

    client.login().block()

    client.eventDispatcher.on(Message::class.java)
        .map { it.content }
        .filter { it.startsWith("!") }
        .flatMap { command ->
            val commandName = command.substring(1)
            when (commandName) {
                "ping" -> handlePingCommand()
                "hello" -> handleHelloCommand()
                "roll" -> handleRollCommand()
                else -> handleUnknownCommand(commandName)
            }
        }
        .flatMap { response -> client.eventDispatcher.emit(response) }
        .onErrorResume { error ->
            println("Error: ${error.message}")
            client.logout().block()
            client.login().block()
        }
        .subscribe()
}

fun handlePingCommand(): String {
    return "Pong!"
}

fun handleHelloCommand(): String {
    return "Hello, world!"
}

fun handleRollCommand(): String {
    val random = (1..6).random()
    return "You rolled a $random!"
}

fun handleUnknownCommand(commandName: String): String {
    return "Unknown command: $commandName"
}
